<?xml version="1.0" encoding="utf-8"?>
<launch>
  <param name="/agentNum" value="1" />
  <rosparam command="load" file="$(find task_switch)/config/xyztheta.yaml" />

  <include file="$(find task_switch)/launch/joy.launch" />
  <node name="poseCollector" pkg="task_switch" type="posecollector.py" output="screen">
    <!-- for specify /bebop10*/virtualdrone/posestamped -->
    <!-- "preTopicName"bebop10*/"posestampedTopic" will be subscribed-->
    <param name="posestampedTopic" value="/virtualdrone/posestamped" />
    <param name="preTopicName" value="/" />
  </node>


  <node name="pcc_parameter" pkg="task_switch" type="task_switch_pcc_param_server" output="screen" />
  <node name="charge_parameter" pkg="task_switch" type="task_switch_charge_param_server" output="screen" />
  <node name="cbf_parameter" pkg="task_switch" type="task_switch_cbf_param_server" output="screen" />
  <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" output="screen" />
  <node pkg="task_switch" name="central" type="central_xyztheta.py" output="screen">
    <param name="clock" value="20" />
  </node>
  <!-- Run simulation -->
  <include file="$(find task_switch)/launch/agent_xyztheta.launch">
    <arg name="number" value="1" />
  </include>

  <!-- <include file="$(find task_switch)/launch/agent_xyztheta.launch">
    <arg name="number" value="2" />
  </include> -->

  <!-- <include file="$(find task_switch)/launch/theta_1d_agent.launch">
      <arg name="number" value="3"/>
    </include> -->

  <node pkg="task_switch" name="visualize_surf" type="pointcloud_2d.py" output="screen">
    <remap from="surf_value" to="/info" />

  </node>
  <node pkg="task_switch" name="visualize_theta" type="pointcloud_2d.py" output="screen">
    <remap from="surf_value" to="/theta_phi" />
    <remap from="/rviz_field" to="/rviz_theta_field" />
    <remap from="/point_cloud2" to="/pointcloud_theta" />
  </node>
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find task_switch)/config/sim_python.rviz" />
</launch>